### DISCUSSION TITLE
Thanks this works.

### DISCUSSION THREAD

**Discussion by joelrb** [Hawxy/Clerk.Net#9] [General]
I confirm this works without any changes. Thanks!

I suggest to add also a token validator if we use JWKS.

1. Create a JWT template in Clerk. Use the template in client to get the token: `await getToken({ template: 'apijwt' })` .
2. In .Net download the JWKS from Clerk: `https://<yoururl>.clerk.accounts.dev/.well-known/jwks.json`.
3. Validate the token from 1 with the JWKS file.
4. If valid, allow continue processing API request.

********
`using System;
using System.IdentityModel.Tokens.Jwt;
using Microsoft.IdentityModel.Tokens;
using System.Net.Http;
using System.Threading.Tasks;

class Program
{
    static async Task Main()
    {
        var token = "your_jwt_token";
        var jwksEndpoint = "https://_yoururl.clerk.accounts.dev/.well-known/jwks.json";

        var jsonWebKeySet = await GetJsonWebKeySetAsync(jwksEndpoint);

        var handler = new JwtSecurityTokenHandler();
        var validationParameters = new TokenValidationParameters
        {
            ValidateIssuerSigningKey = true,
            IssuerSigningKeyResolver = (s, securityToken, identifier, parameters) => jsonWebKeySet.FindKey(identifier),
            ValidIssuer = "your_issuer",
            ValidAudience = "your_audience"
        };

        try
        {
            var claimsPrincipal = handler.ValidateToken(token, validationParameters, out _);
            Console.WriteLine("Token is valid.");
        }
        catch (SecurityTokenException)
        {
            Console.WriteLine("Invalid token.");
        }
    }

    static async Task<JsonWebKeySet> GetJsonWebKeySetAsync(string jwksEndpoint)
    {
        using (var httpClient = new HttpClient())
        {
            var json = await httpClient.GetStringAsync(jwksEndpoint);
            return new JsonWebKeySet(json);
        }
    }
}`

**Comment by Hawxy** [comment]
@joelrb ASP.NET Core authentication middleware automatically downloads and validates JWKS. Doing it yourself isn't required.
